src/tasks/publish/index.ts

```diff
@@ -149,7 +152,11 @@ async function publishTask(
   // So users do not have to wait a long time before seeing the config is not okay
   if (!rootAccount)
     throw new BuidlerPluginError(
-      `No account configured. Provide a mnemonic or private key for ${selectedNetwork} in `
+      `No account configured. Provide a mnemonic or private key for ${selectedNetwork} in the buidler.config.json. For more information check: https://buidler.dev/config/#json-rpc-based-networks`
+    )
+  if (verify && !hasEtherscanKey)
```

Comment:
Add an extra sanity check if the user decide to verify the contracts using Etherscan.